### PR TITLE
Prime thumbnail cache.

### PR DESCRIPTION
**Pull request by spacedmonkey**
Trac ticket: https://core.trac.wordpress.org/ticket/55592

---
**This Pull Request is for code review only. Please keep all other discussion in the Trac ticket. Do not merge this Pull Request. See [GitHub Pull Requests for Code Review](https://make.wordpress.org/core/handbook/contribute/git/github-pull-requests-for-code-review/) in the Core Handbook for more details.**

### CODE REVIEW

File: tests/phpunit/tests/rest-api/rest-posts-controller.php [hunk 1]
```diff
@@ -1517,6 +1517,47 @@ public function test_get_items_not_sticky_with_exclude_no_sticky_posts() {
 		$this->assertPostsWhere( " AND {posts}.ID NOT IN ($id3) AND {posts}.post_type = 'post' AND (({posts}.post_status = 'publish'))" );
 	}
 
+	/**
+	 * @ticket 55592
+	 * @covers WP_REST_Posts_Controller::get_items()
+	 */
+	public function test_get_items_with_featured_media() {
```
Comment (peterwilsoncc) [inline tests/phpunit/tests/rest-api/rest-posts-controller.php:1521]: This test is failing in a bunch of places, are you able to take a look as to why?

Comment (manuelRod) [reply tests/phpunit/tests/rest-api/rest-posts-controller.php:1521]: @spacedmonkey maybe is the order of $attachment_ids.

Comment (spacedmonkey) [reply tests/phpunit/tests/rest-api/rest-posts-controller.php:1521]: Tests are now passing.

File: tests/phpunit/tests/rest-api/rest-posts-controller.php [hunk 2]
```diff
@@ -1517,6 +1517,41 @@ public function test_get_items_not_sticky_with_exclude_no_sticky_posts() {
...
+		$file           = DIR_TESTDATA . '/images/canola.jpg';
+		$attachment_ids = array();
+		$post_ids       = array();
+		for ( $i = 0; $i < 3; $i ++ ) {
```
Comment (mitogh) [inline tests/phpunit/tests/rest-api/rest-posts-controller.php]:
```suggestion
		for ( $i = 0; $i < 3; $i++ ) {
```

File: tests/phpunit/tests/rest-api/rest-posts-controller.php [hunk 3]
```diff
@@ -1517,6 +1517,41 @@ public function test_get_items_not_sticky_with_exclude_no_sticky_posts() {
...
+		$filter = new MockAction();
+		add_filter( 'update_post_metadata_cache', array( $filter, 'filter' ), 10, 2 );
+
+		$request = new WP_REST_Request( 'GET', '/wp/v2/posts' );
+		$request->set_param( 'include', $post_ids );
+		rest_get_server()->dispatch( $request );
+
+		$args = $filter->get_args();
+		$last = end( $args );
+		$this->assertEqualSets( $attachment_ids, $last[1] );
```
Comment (mitogh) [inline tests/phpunit/tests/rest-api/rest-posts-controller.php:1550]: I think before accessing the index of `last` would be worth checking that is in fact an array not `null`.

```suggestion
		$this->assertIsArray( $last );
		$this->assertEqualSets( $attachment_ids, $last[1] );
```

**Comment by spacedmonkey** [comment]
Committed in https://core.trac.wordpress.org/changeset/53499

**Pull request closed without merging**